### PR TITLE
added creation, deletion and get custom or environment domains

### PR DESCRIPTION
**Pull request by marcosOrdieres**

### CODE REVIEW

File: src/model/Project.ts [hunk 1]
```diff
@@ -211,6 +214,61 @@ export default class Project extends Ressource {
...
+    replacement_for?: string,
+    ssl = []
+  ) {
+    const body: Partial<EnvironmentDomain> = { name, type, replacement_for };
+
+    if (ssl.length) {
+      body.ssl = ssl;
+    }
+
+    if (replacement_for) {
```
Comment (shakaman) [inline src/model/Project.ts:255]:
```suggestion
    if (replacement_for) {
          body.replacement_for = replacement_for;
    } else {
          body.is_default = true;
    }
```

File: src/model/Project.ts [hunk 2]
```diff
@@ -211,6 +214,61 @@ export default class Project extends Ressource {
...
+   * @param array  ssl
+   *
+   * @return Result
+   */
+  async addEnvironmentDomain(
+    name: string,
+    environmentId: string,
+    type = "replacement",
+    replacement_for?: string,
+    ssl = []
```
Comment (shakaman) [inline src/model/Project.ts:247]: It should be cleaner to have:
```suggestion
    ssl?: Array<string>
```

After you can have:
```js
    if (ssl) {
      body.ssl = ssl;
    }
```

Comment (marcosOrdieres) [reply src/model/Project.ts]: Unfortunately, If I do like this there is a Typescript error in here

File: src/model/Project.ts [hunk 3]
```diff
@@ -211,6 +214,63 @@ export default class Project extends Ressource {
...
+   * @param array  ssl
+   *
+   * @return Result
+   */
+  async addEnvironmentDomain(
+    name: string,
+    environmentId: string,
+    type = "replacement",
+    replacement_for?: string,
+    ssl?: never[] | undefined
```
Comment (shakaman) [inline src/model/Project.ts:247]: if you use `?` you don't need to explicitly define `undefined`

**Pull request merged**